src/iuse.cpp

```diff
@@ -8083,6 +8085,324 @@ std::optional<int> iuse::capture_monster_act( Character *p, item *it, const trip
+{
+    if( p->cant_do_mounted() ) {
+        return std::nullopt;
+    }
+    p->inv->restack( *p );
+    item_location heater = item_location( *p, it );
+    int fire_flag = 0;
+    //Hotplate can only use it self as heat source
+    if( get_map().has_nearby_fire( p->pos() ) && !it->has_quality( qual_HOTPLATE ) ) {
+        p->add_msg_if_player( m_info, _( "You put %1$s on fire to start heating." ), it->tname() );
+        fire_flag = 1;
+    } else if( it->has_quality( qual_HOTPLATE ) ) {
+        if( it->ammo_remaining() >= it->type->charges_to_use() ) {
+            heater = item_location( *p, it );
+            p->add_msg_if_player( m_info, _( "You use %1$s to start heating." ), heater->tname() );
+        } else if( !it->has_no_links() ) {
+            heater = item_location( *p, it );
+            p->add_msg_if_player( m_info, _( "You use %1$s to start heating." ), heater->tname() );
+        } else if( it->has_flag( flag_USE_UPS ) &&
+                   units::to_kilojoule( p->available_ups() ) >= it->type->charges_to_use() ) {
+            heater = item_location( *p, it );
+            p->add_msg_if_player( m_info, _( "You use %1$s to start heating." ), heater->tname() );
+        } else {
+            p->add_msg_if_player( m_info, _( "The %s's batteries are dead." ), it->tname() );
+            return std::nullopt;
+        }
+    } else if( !it->has_quality( qual_HOTPLATE ) ) {
+        auto filter = [p]( const item & e ) {
+            if( e.has_quality( qual_HOTPLATE, 2 ) && e.ammo_remaining() >= e.type->charges_to_use() ) {
+                return true;
+            }
+            if( e.has_quality( qual_HOTPLATE, 2 ) && ( !e.has_no_links() ) ) {
+                if( e.link().t_veh->connected_battery_power_level().first >= e.type->charges_to_use() ) {
+                    return true;
+                }
+            }
+            if( e.has_quality( qual_HOTPLATE, 2 ) && e.has_flag( flag_USE_UPS ) ) {
+                if( units::to_kilojoule( p->available_ups() ) >= e.type->charges_to_use() ) {
+                    return true;
+                }
+            }
+            return false;
+        };
+        heater = g->inv_map_splice( filter, _( "Select a tool to heat:" ), 1,
+                                    _( "You don't have proper heating source." ) );
+        if( !heater ) {
+            add_msg( m_info, _( "Never mind." ) );
+            return std::nullopt;
+        }
+        p->add_msg_if_player( m_info, _( "You put %1$s on %2$s to start heating." ), it->tname(),
+                              heater->tname() );
+    }
+    if( heat_item( *p ) ) {
+        if( fire_flag != 1 ) {
+            heater->activation_consume( 1, heater.position(), p );
+        }
+    }
+    return 0;
+}
+
+std::optional<int> iuse::heat_items( Character *p, item *it, bool liquid_items, bool solid_items )
+{
+    if( p->cant_do_mounted() ) {
+        return std::nullopt;
+    }
```

Comment:
There's no reason for this to be done in the helper functions instead of the iuse.